src/reporter.js

```diff
@@ -197,6 +204,26 @@ function CoverageIstanbulReporter(baseReporterDecorator, logger, config) {
       });
     });
 
+    if (
+      Object.keys(temporaryFileDateResult).length > 0 &&
+      coverageConfig.tempDirectory
+    ) {
+      const tempFolder = coverageConfig.tempDirectory;
+
+      if (fs.existsSync(tempFolder) === true) {
+        fs.removeSync(tempFolder);
```

Comment:
Doing this will wipe the temp directory on every run, so from what I can tell only one coverage file will ever be written there and when you run `nyc merge .nyc_output coverage.json` it won't include output files from all runs. 

I guess we also want an option to control whether to delete the temp directory, so you could set it to delete it only on the first repo that runs and then subsequent repos in your mono repo wouldn't delete it. This is kind of how the [nyc example](https://github.com/istanbuljs/nyc#combining-reports-from-multiple-runs) works, where `cover:integration` uses `--no-clean` to not wipe the output directory before running.

WDYT?